src/main/java/io/odpf/depot/bigquery/json/BigqueryJsonUpdateListener.java

```diff
@@ -1,24 +1,98 @@
 package io.odpf.depot.bigquery.json;
 
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.Schema;
+import io.odpf.depot.bigquery.handler.BigQueryClient;
 import io.odpf.depot.bigquery.handler.MessageRecordConverter;
 import io.odpf.depot.bigquery.handler.MessageRecordConverterCache;
+import io.odpf.depot.bigquery.proto.BigqueryFields;
+import io.odpf.depot.common.TupleString;
 import io.odpf.depot.config.BigQuerySinkConfig;
 import io.odpf.depot.message.OdpfMessageParser;
 import io.odpf.depot.stencil.OdpfStencilUpdateListener;
 
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
 public class BigqueryJsonUpdateListener extends OdpfStencilUpdateListener {
     private final MessageRecordConverterCache converterCache;
     private BigQuerySinkConfig config;
```

Comment:
can we make `config` Final.

Reply:
will do,  but we should have some checkstyle rule to automatically detect and enforce it.